course03/project01/balint/application.cc

```diff
@@ -0,0 +1,257 @@
+#include "cinder/app/App.h"
+#include "cinder/app/RendererGl.h"
+#include "cinder/gl/gl.h"
+
+using namespace ci;
+using namespace ci::app;
+
+// We'll create a new Cinder Application by deriving from the App class.
+class BasicApp : public App
+{
+ public:
+  // Cinder will call 'keyDown' when the user presses a key on the keyboard.
+  // See also: keyUp.
+  void keyDown(KeyEvent event) override;
+
+  // Cinder will call 'draw' each time the contents of the window need to be redrawn.
+  void draw() override;
+  void update() override;
+  void setup() override;
+
+ private:
+  // This will maintain a list of points which we will draw line segments between
+  float mBarrierWidth = 5;
+  vec2 mBallPos;
+  vec2 mBallVel;
+  float mBatLength = 200;
+  float mBallRadius = 13;
+  float height = 600;
+  float width = 800;
+
+  void drawSurroundings();
+  void drawBall();
+  void drawLeftBat();
+  void drawRightBat();
+  void moveBall();
+
+  Rectf mLeftBarrier;
+  Rectf mRightBarrier;
+  Rectf mTopBarrier;
+  Rectf mBottomBarrier;
+
+  Rectf mLeftBat, mRightBat;
+
+  bool mGameOver = false;
+
+  bool ballCollidesDown_p() { return ballCollidesWithRect(mBottomBarrier); }
+  bool ballCollidesRight_p() { return ballCollidesWithRect(mRightBarrier); }
+  bool ballCollidesUp_p() { return ballCollidesWithRect(mTopBarrier); }
+  bool ballCollidesLeft_p() { return ballCollidesWithRect(mLeftBarrier); }
+
+  bool ballCollidesWithRect(Rectf& rect);
+  vec2 advanceBall();
+  void moveLeftBat(bool up_p);
+  void moveRightBat(bool up_p);
+};
+
+void prepareSettings(BasicApp::Settings* settings)
+{
+  settings->setMultiTouchEnabled(false);
+  settings->setWindowSize(800, 600);
+}
+
+void BasicApp::update()
+{
+  static const double timestep = 1.0 / 60.0;
```

Comment:
A good practice is to either use strong type (e.g. a type Seconds or TimeSpan). This can add some burden when using the type, but at the very least you should include the unit in the name of the variable: `timestepSeconds`. It reduces the chance of having the variable misused / people thinking that it is stored in a different unit.